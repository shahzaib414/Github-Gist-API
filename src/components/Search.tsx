import React, { ChangeEvent, useState } from 'react';
import styled from "styled-components";
import { TextField, Button } from "@material-ui/core"

const StyledDiv = styled.div`
justify-content: center;
display: flex;

    .MuiFormControl-root,
    .MuiTextField-root {
        width: 50%;
    }

    .MuiButton-containedPrimary {
        margin-left: 20px;
    }
`;

interface Props {
    searchPlaceHolder?: string;
    onClick: (value: string) => void;
    searchButtonText?: string
}

const Search = ({ searchButtonText, searchPlaceHolder, onClick }: Props) => {
    const [searchQuery, setSearchQuery] = useState<string>('');

    const placeholder = searchPlaceHolder || 'Search';

    const onTextChangeHanlder = (value: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
        setSearchQuery(value.target.value)
    };
    return <StyledDiv>
        <TextField id="standard-basic" label={placeholder} value={searchQuery} onChange={onTextChangeHanlder} />
        <Button variant="contained" color="primary" onClick={() => onClick(searchQuery)}>
            {searchButtonText || `Search`}
        </Button>
    </StyledDiv>
}

export default Search;