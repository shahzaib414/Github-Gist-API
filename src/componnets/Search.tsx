import React, { ChangeEvent, useState } from 'react';
import styled from "styled-components";
import { TextField, Button } from "@material-ui/core"

const Container = styled.div`
justify-content: center;
display: flex;

    .MuiFormControl-root,
    .MuiTextField-root {
        width: 50%;
    }

    .MuiButton-containedPrimary {
        margin-left: 20px;
    }
`;

interface Props {
    searchPlaceHolder?: string;
    onClick: (value: string) => void;
    searchButtonText?: string
}

const Search = ({ searchButtonText, searchPlaceHolder, onClick }: Props) => {
    const [searchQuery, setSearchQuery] = useState<string>('');

    const placeholder = searchPlaceHolder || 'GitHub username';

    const onTextChangeHanlder = (value: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
        setSearchQuery(value.target.value)
    };
    return <Container>
        <TextField id="standard-basic" label={placeholder} value={searchQuery} onChange={onTextChangeHanlder}/>
        <Button variant="contained" color="primary" onClick={() => onClick(searchQuery)}>
            {searchButtonText || `Search`}
        </Button>
    </Container>
}

export default Search;