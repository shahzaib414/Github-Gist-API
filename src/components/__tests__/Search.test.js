import React from 'react';
import renderer from 'react-test-renderer';
import Search from "../Search";

test("Search Component renders correctly", () => {
    const component = renderer.create(<Search onClick={() => {}}/>)
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})
