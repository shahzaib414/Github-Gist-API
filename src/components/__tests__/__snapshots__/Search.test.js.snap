// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component renders correctly 1`] = `
<div
  className="sc-bdvvaa fQTtyz"
>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
      htmlFor="standard-basic"
      id="standard-basic-label"
    >
      Search
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="standard-basic"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Search
    </span>
  </button>
</div>
`;
